### PR TITLE
run simple tests on windows and linux

### PR DESCRIPTION
**Pull request by Emilgardis**
solves parts of #610, depends on #852

### CODE REVIEW

**Comment by Emilgardis** [comment]
interesting, this all works for me, here the run fails

File: src/docker/shared.rs [hunk 1]
```diff
@@ -339,29 +339,61 @@ pub(crate) fn docker_mount(
     Ok(mount_volumes)
 }
 
+#[cfg(target_os = "windows")]
+fn winpath(path: &Path, verbose: bool) -> Result<PathBuf> {
```
Comment (Alexhuszagh) [inline src/docker/shared.rs:343]: I believe this was added to this commit by mistake?

Comment (Emilgardis) [reply src/docker/shared.rs:343]: no, see the previous error in https://github.com/cross-rs/cross/runs/7045847730?check_suite_focus=true

**Comment by Emilgardis** [comment]
on windows, getting

```
rs.for-cross-target=x86_64-unknown-linux-gnu' --label 'org.cross-rs.workspace_root=C:\Users\runneradmin\AppData\Local\Temp\tmp.AExM7t25b6\workspace' --tag cross-custom-workspace:x86_64-unknown-linux-gnu-9f829-pre-build --build-arg 'CROSS_CMD=apt-get update && apt-get install -y gcc' --build-arg 'CROSS_DEB_ARCH=amd64' --file 'C:\Users\runneradmin\AppData\Local\Temp\tmp.AExM7t25b6\workspace\target\x86_64-unknown-linux-gnu\Dockerfile.x86_64-unknown-linux-gnu-custom' .
Sending build context to Docker daemon    170kB
Error response from daemon: unsupported os linux
Error: 
   0: when building custom image
   1: when pre-building
   2: `"C:\Program Files\Docker\docker.EXE" build --platform linux/amd64 --label 'org.cross-rs.for-cross-target=x86_64-unknown-linux-gnu' --label 'org.cross-rs.workspace_root=C:\Users\runneradmin\AppData\Local\Temp\tmp.AExM7t25b6\workspace' --tag cross-custom-workspace:x86_64-unknown-linux-gnu-9f829-pre-build --build-arg 'CROSS_CMD=apt-get update && apt-get install -y gcc' --build-arg 'CROSS_DEB_ARCH=amd64' --file 'C:\Users\runneradmin\AppData\Local\Temp\tmp.AExM7t25b6\workspace\target\x86_64-unknown-linux-gnu\Dockerfile.x86_64-unknown-linux-gnu-custom' .` failed with exit code: 1
Note: CROSS_CMD=apt-get update && apt-get install -y gcc
Error: Process completed with exit code 1.
```

Not sure how to enable it properly.

**Comment by Emilgardis** [comment]
this is impossible, see https://github.com/actions/virtual-environments/issues/2216

**Pull request closed without merging**